File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/models/investigation/investigationmodels/investigationtype/evidence/EvidenceListModel.java
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.models.investigation.investigationmodels.investigationtype.evidence;

import android.content.Context;
import android.content.res.TypedArray;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.TritiumGaming.phasmophobiaevidencepicker.R;

import java.util.ArrayList;

public class EvidenceListModel {

    @Nullable
    public static ArrayList<EvidenceModel> evidenceList = null;

    public void init(@NonNull Context c) {
        evidenceList = new ArrayList<>();
        String[] evidenceNames = c.getResources().getStringArray(R.array.evidence_type_names);
        TypedArray typedArray = c.getResources().obtainTypedArray(R.array.evidence_icon_array);
        for (int i = 0; i < evidenceNames.length; i++) {
            EvidenceModel evidence = new EvidenceModel();
            evidence.setName(evidenceNames[i]);
            evidence.setIcon(typedArray.getResourceId(i, 0));
            evidenceList.add(evidence);
        }
        typedArray.recycle();
    }

    public static int getCount() {
        if(evidenceList == null) return 0;
        return evidenceList.size();
    }

    public ArrayList<EvidenceModel> getList() {
        return evidenceList;
    }

    /**
     * Resets the Ruling for each Evidence type
     */
    public void reset() {
        for (int i = 0; i < evidenceList.size(); i++) {
            evidenceList.get(i).setRuling(EvidenceModel.Ruling.NEUTRAL);
        }
    }
}

File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/models/investigation/investigationmodels/investigationtype/evidence/EvidenceListModel.kt
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.models.investigation.investigationmodels.investigationtype.evidence

import android.content.Context
import com.TritiumGaming.phasmophobiaevidencepicker.R

class EvidenceListModel {

    val evidenceList: ArrayList<EvidenceModel> = ArrayList(0)

    val count: Int
        get() = evidenceList.size


    fun init(c: Context) {
        evidenceList.clear()
        val evidenceNames = c.resources.getStringArray(R.array.evidence_type_names)
        val typedArray = c.resources.obtainTypedArray(R.array.evidence_icon_array)
        for (i in evidenceNames.indices) {
            val evidence = EvidenceModel()
            evidence.name = evidenceNames[i]
            evidence.icon = typedArray.getResourceId(i, 0)
            evidenceList?.add(evidence)
        }
        typedArray.recycle()
    }

    /** Resets the Ruling for each Evidence type */
    fun reset() {
        for (i in evidenceList.indices) {
            evidenceList[i].ruling = EvidenceModel.Ruling.NEUTRAL
        }
    }

}
